include/zio/zio_fifo_buf.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019 Thomas Burdick <thomas.burdick@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file zio_fifo_buf.h
+ *
+ * @brief ZIO FIFO Buffer provides a software zio_buf implementation
+ * based on a zio_fifo
+ */
+
+#ifndef ZEPHYR_INCLUDE_ZIO_FIFO_BUF_H_
+#define ZEPHYR_INCLUDE_ZIO_FIFO_BUF_H_
+
+#include <zio/zio_fifo.h>
+#include <zio/zio_buf.h>
+
+/**
+ * @private
+ * @brief Software FIFO implementation of a zio_buf
+ */
+struct z_zio_fifo_buf {
+	u32_t watermark;
+	u32_t length;
+	struct k_poll_signal signal;
```

Comment:
Making this a pollable object will reduce size of this struct and simplify its usage.

Reply:
Indeed, I was considering changing the kernel.h to include this type to bind a k_poll_event to, but this was an easy way to hash out the idea

Reply:
Thinking more about this I think this is the right way to do it, I didn't want to dig too deep into how to make it a pollable object just yet. I'd say that would be something to fix before the topic branch gets merged.